fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/LambdaFunctionOperator.java

```diff
@@ -0,0 +1,100 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.operator.scalar;
+
+import com.google.common.collect.Lists;
+import com.starrocks.catalog.Type;
+import com.starrocks.sql.optimizer.base.ColumnRefSet;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+
+import java.util.List;
+import java.util.Objects;
+
+public class LambdaFunctionOperator extends ScalarOperator {
+
+    private List<ColumnRefOperator> refColumns;
+    private ScalarOperator lambdaExpr;
+
+    public LambdaFunctionOperator(List<ColumnRefOperator> refColumns, ScalarOperator lambdaExpr, Type retType) {
+        super(OperatorType.LAMBDA_FUNCTION, retType);
+        this.refColumns = refColumns;
+        this.lambdaExpr = lambdaExpr;
+    }
+
+    public List<ColumnRefOperator> getRefColumns() {
+        return refColumns;
+    }
+
+    public ScalarOperator getLambdaExpr() {
+        return lambdaExpr;
+    }
+
+    // only need to concern the lambda expression.
+    @Override
+    public List<ScalarOperator> getChildren() {
+        return Lists.newArrayList(lambdaExpr);
+    }
+
+    @Override
+    public ScalarOperator getChild(int index) {
+        return lambdaExpr;
+    }
+
+
+    @Override
+    public boolean isNullable() {
+        return lambdaExpr.isNullable();
+    }
+
+    @Override
+    public void setChild(int index, ScalarOperator child) {
```

Comment:
should add a         Preconditions.checkState(false) here.

Reply:
ok